base/file.jl

```diff
@@ -341,37 +353,46 @@ Obtain the path of a temporary directory (possibly shared with other processes).
 tempdir()
 
 """
-    tempname()
+    tempname(; prefix="$temp_prefix")
 
-Generate a unique temporary file path.
+Generate a unique temporary file path. A `prefix` can be provided in addition, which
+otherwise defaults to "$temp_prefix".
+
+*Note: The `prefix` may be truncated based on the limitations of the OS.*
 """
-tempname()
+tempname(; prefix=temp_prefix)
 
 """
-    mktemp(parent=tempdir())
+    mktemp(parent=tempdir(); prefix="$temp_prefix")
 
 Returns `(path, io)`, where `path` is the path of a new temporary file in `parent` and `io`
-is an open file object for this path.
+is an open file object for this path. A `prefix` can be provided in addition, which
```

Comment:
As I've noted several times, please do not repeat the default, which is already shown in the signature. This applies to other docstrings as well.